test/params_api_test.c

```diff
@@ -36,16 +36,6 @@ static void copy_to_le(unsigned char *out, const void *in, size_t len)
         swap_copy(out, in, len);
 }
 
```

Comment:
It still seems it is still broken with zeros in the incorrect place...

    # Subtest: test_param_long
    1..6
    # ERROR: (memory) 'cmp == raw_values[n].value' failed @ ../test/params_api_test.c:166
    # --- cmp
    # +++ raw_values[n].value
    # 0000:-00000000
    # 0000:+3827bf3b
    #       ^^^^^^^^
    # 
    not ok 1 - iteration 1
    # ERROR: (memory) 'cmp == raw_values[n].value' failed @ ../test/params_api_test.c:166
    # --- cmp
    # +++ raw_values[n].value
    # 0000:-00000000
    # 0000:+9f264822
    #       ^^^^^^^^
    # 
    not ok 2 - iteration 2
    ok 3 - iteration 3
    ok 4 - iteration 4
    ok 5 - iteration 5
    ok 6 - iteration 6
not ok 2 - test_param_long

    # Subtest: test_param_ulong
    1..6
    # ERROR: (memory) 'cmp == raw_values[n].value' failed @ ../test/params_api_test.c:221
    # --- cmp
    # +++ raw_values[n].value
    # 0000:-00000000
    # 0000:+3827bf3b
    #       ^^^^^^^^
    # 
    not ok 1 - iteration 1
    # ERROR: (memory) 'cmp == raw_values[n].value' failed @ ../test/params_api_test.c:221
    # --- cmp
    # +++ raw_values[n].value
    # 0000:-00000000
    # 0000:+9f264822
    #       ^^^^^^^^
    # 
    not ok 2 - iteration 2
    ok 3 - iteration 3
    ok 4 - iteration 4
    ok 5 - iteration 5
    ok 6 - iteration 6
not ok 4 - test_param_ulong
    # Subtest: test_param_size_t
    1..6
    # ERROR: (memory) 'cmp == raw_values[n].value' failed @ ../test/params_api_test.c:361
    # --- cmp
    # +++ raw_values[n].value
    # 0000:-00000000
    # 0000:+3827bf3b
    #       ^^^^^^^^
    # 
    not ok 1 - iteration 1
    # ERROR: (memory) 'cmp == raw_values[n].value' failed @ ../test/params_api_test.c:361
    # --- cmp
    # +++ raw_values[n].value
    # 0000:-00000000
    # 0000:+9f264822
    #       ^^^^^^^^
    # 
    not ok 2 - iteration 2
    ok 3 - iteration 3
    ok 4 - iteration 4
    ok 5 - iteration 5
    ok 6 - iteration 6
not ok 7 - test_param_size_t
    # Subtest: test_param_int64
    1..6
    # ERROR: (memory) 'cmp == raw_values[n].value' failed @ ../test/params_api_test.c:305
    # --- cmp
    # +++ raw_values[n].value
    # 0000:-00000000
    # 0000:+3827bf3b
    #       ^^^^^^^^
    # 
    not ok 1 - iteration 1
    # ERROR: (memory) 'cmp == raw_values[n].value' failed @ ../test/params_api_test.c:305
    # --- cmp
    # +++ raw_values[n].value
    # 0000:-00000000
    # 0000:+9f264822
    #       ^^^^^^^^
    # 
    not ok 2 - iteration 2
    ok 3 - iteration 3
    ok 4 - iteration 4
    ok 5 - iteration 5
    ok 6 - iteration 6
not ok 8 - test_param_int64
    # Subtest: test_param_uint64
    1..6
    # ERROR: (memory) 'cmp == raw_values[n].value' failed @ ../test/params_api_test.c:333
    # --- cmp
    # +++ raw_values[n].value
    # 0000:-00000000
    # 0000:+3827bf3b
    #       ^^^^^^^^
    # 
    not ok 1 - iteration 1
    # ERROR: (memory) 'cmp == raw_values[n].value' failed @ ../test/params_api_test.c:333
    # --- cmp
    # +++ raw_values[n].value
    # 0000:-00000000
    # 0000:+9f264822
    #       ^^^^^^^^
    # 
    not ok 2 - iteration 2
    ok 3 - iteration 3
    ok 4 - iteration 4
    ok 5 - iteration 5
    ok 6 - iteration 6
not ok 9 - test_param_uint64